akka-http-core/src/main/scala/akka/http/impl/util/One2OneBidiFlow.scala

```diff
@@ -87,8 +91,13 @@ private[http] object One2OneBidiFlow {
           } else throw new UnexpectedOutputException(element)
         }
         override def onUpstreamFinish(): Unit = {
-          if (pending == 0 && isClosed(inIn) && !innerFlowCancelled) complete(outOut)
-          else throw OutputTruncationException
+          if (pending > 0 || innerFlowCancelled) {
+            if (materializer.isInstanceOf[ActorMaterializer])
+              ActorMaterializer.downcast(materializer).logger.debug(
+                "Inner stream finished before inputs completed. " +
```

Comment:
Hmm, thinking about it, isn't the case that `(pending > 0 && innerFlowCompleted)` is definitely a truncation (and the only one that is reliable)? Since that means that we passed something to the inner flow which will never return as it has completed. Cancellation means nothing since we already passed it into the inner flow when we increased pending see https://github.com/akka/akka/pull/20128/files#diff-7572c326b2938b0c69732c6f12f643f2R65